File: Data Structures & Algorithms/sliding-window-maximum/submission-10.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:

        dq = collections.deque()
        output = []

        for i, v in enumerate(nums):
            while dq and v > nums[dq[-1]]:
                dq.pop()
            dq.append(i)

            if dq[0] <= i - k:
                dq.popleft()
            
            if i + 1 >= k:
                output.append(nums[dq[0]])
        
        return output

File: Data Structures & Algorithms/two-integer-sum/submission-10.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut mp = HashMap::new();

        for (i, num) in nums.iter().enumerate() {
            let diff = target - num;
            if let Some(ans) = mp.get(&diff) {
                return vec![*ans as i32, i as i32];
            };
            mp.insert(*num, i);
        };
        vec![]
    }
}
